client/data/external-contributors/use-external-contributors.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import { useQuery } from 'react-query';
+/**
+ * Internal dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+
+const useExternalContributorsQuery = ( siteId, queryOptions = {} ) => {
+	return useQuery(
```

Comment:
It's possible that the `siteId` is null. The previous code guarded against that by calling the `requestExternalContributors` conditionally.

In React Query, we handle this condition by adding an `enabled: !! siteId` option to `useQuery`. See for example the `useUsersQuery` hook.